Jenkinsfile

```diff
@@ -0,0 +1,247 @@
+@Library("rd-apmm-groovy-ci-library@v1.x") _
+
+/*
+ Runs the following steps in parallel and reports results to GitHub:
+ - Lint using flake8
+ - Run Python 2.7 unit tests in tox
+ - Run Python 3 unit tests in tox
+ - Build Debian packages for supported Ubuntu versions
+
+ If these steps succeed and the master branch is being built, wheels and debs are uploaded to Artifactory and the
+ R&D Debian mirrors.
+
+ Optionally you can set FORCE_PYUPLOAD to force upload to Artifactory, and FORCE_DEBUPLOAD to force Debian package
+ upload on non-master branches.
+*/
+
+pipeline {
+    agent {
+        label "16.04&&ipstudio-deps"
+    }
+    options {
+        ansiColor('xterm') // Add support for coloured output
+        buildDiscarder(logRotator(numToKeepStr: '10')) // Discard old builds
+    }
+    parameters {
+        booleanParam(name: "FORCE_PYUPLOAD", defaultValue: false, description: "Force Python artifact upload")
+        booleanParam(name: "FORCE_DEBUPLOAD", defaultValue: false, description: "Force Debian package upload")
+    }
+    environment {
+        http_proxy = "http://www-cache.rd.bbc.co.uk:8080"
+        https_proxy = "http://www-cache.rd.bbc.co.uk:8080"
+    }
+    stages {
+        stage("Clean Environment") {
+            steps {
+                sh 'git clean -df'
```

Comment:
I know this also affects the example, but do we want a `-x` as well? See https://git-scm.com/docs/git-clean#git-clean--x

Reply:
Yes, I think we do. I'll update.

Reply:
I'm happy to make the corresponding change to the template - my bad review didn't catch it to start with!